File: Solutions/_125.go
func isPalindrome(s string) bool {
	s = strings.ToLower(s)
	lPointer := 0
	rPointer := len(s) - 1

	for lPointer < rPointer {
		for !isAlphaNumeric(s[lPointer]) && lPointer < rPointer {
			lPointer++
		}

		for !isAlphaNumeric(s[rPointer]) && lPointer < rPointer {
			rPointer--
		}

		fmt.Printf("%v - %v", lPointer, rPointer)
		if s[lPointer] != s[rPointer] {
			return false
		}

		lPointer++
		rPointer--
	}
	return true
}

func isAlphaNumeric(c byte) bool {

	if (c >= 'a' && c <= 'z') || c >= 'A' && c <= 'Z' || (c >= '0' && c <= '9') {
		return true
	}

	return false
}

File: Solutions/_125.cpp
#include <iostream>
#include <string>

using namespace std;

bool isAlphaNumeric(char c) {
  if (c >= 'a' && c <= 'z')
    return true;
  if (c >= 'A' && c <= 'Z')
    return true;
  if (c >= '0' && c <= '9')
    return true;
  return false;
}

bool isPalindrome(string s) {
  int lPointer = 0;
  int rPointer = s.length() - 1;

  while (lPointer < rPointer) {
    if (!isAlphaNumeric(s[lPointer])) {
      lPointer++;
      continue;
    }
    if (!isAlphaNumeric(s[rPointer])) {
      rPointer--;
      continue;
    }
    if (tolower(s[lPointer]) != tolower(s[rPointer]))
      return false;
    lPointer++;
    rPointer--;
  }
  return true;
}

int main() {
  string input = "A man, a plan, a canal: Panama";
  cout << isPalindrome(input) << endl;
  input = "race a car";
  cout << isPalindrome(input) << endl;
  input = " ";
  cout << isPalindrome(input) << endl;
  return 0;
}
